plugins/ccgx/ccgx-dock.quirk

```diff
@@ -0,0 +1,102 @@
+#
+# Copyright (C) 2020 Cypress Semiconductor Corporation.
+#
+# SPDX-License-Identifier: LGPL-2.1+
+#
+
+#######################################################
+## Gen2 Dock
+#######################################################
+[DeviceInstanceId=USB\VID_17EF&PID_A38F]
+Name = Lenovo Gen2 Dock Billboard
+Plugin = ccgx
+GType = FuCcgxDockBb
+ParentGuid = USB\VID_17EF&PID_A392
+
+[DeviceInstanceId=USB\VID_04B4&PID_521A]
+Name = Lenovo Gen2 Dock USB Serial Bridge
+Plugin = ccgx
+GType = FuCcgxDockBb
+ParentGuid = USB\VID_17EF&PID_A392
+
+# Billboard Controller for Gen2 Dock
+[Guid=USB\VID_17EF&PID_A38F&bb]
+Name = Gen2 Dock Billboard
+Summary = USB-C Docking Billboard
+Plugin = ccgx
+GType = FuCcgxDockBb
+Vendor = Lenovo Inc
+Flags = require-ac,cy-device-external-bb
+
+# CCG PD IC Controller for Gen2 Dock
+[Guid=USB\VID_04B4&PID_521A&bb]
+Name = CCGx PD Controller
+Summary = USB-C PD controller
+Plugin = ccgx
+GType = FuCcgxDockBb
+Vendor = Cypress Semiconductor
+FirmwareSizeMin  = 0x000222
+FirmwareSizeMax = 0x042420
+InstallDuration = 60
+Flags = require-ac,dual-image,self-recovery,cy-device-pd-i2c,cy-fw-image-dual-asymmetric
+DeviceSiliconID = 0x1F00
+DeviceFWAppType = 0x6D64
+ModelName = Gen2
+UpdateMessage = Firmware update will complete after the dock reboot. The dock will not be available for approximately 1 minute.
```

Comment:
In this context, what's `dock reboot`? Would restarting the computer that's attached to the dock be enough?

Reply:
User does not needs to do anything here. Gen2 dock is powered through Cypress device. Device firmware update on cypress dual fw image device requires resetting the device twice: one reset after updating each image.  

This message is just for information that the dock will not be functional till cypress device completes boot up sequence.

Brief overview:
Cypress device has dual firmware image: backup or FW1 and primary FW2 image. Backup image has minimal functionality and mainly used to update primary firmware. To avoid corruption flash area used by active FW is locked. So in order to update FW2, device should be running from FW1 and vice versa.

At a high level: For Gen2 we initially update FW1 by writing new Fw image to FW1 flash area and issue a reset command to device Fw, requesting it to switch Fw image and run from backup fw. 
While switching the FW execution the Cypress device gets reset and the device will not be available till cypress device completes booting up and start executing from backup firmware image.
Now device will be running from FW1 or backup image and we can update FW2 or primary image. After completing FW update to FW2 regin, we again issue a reset request (Jump to alt FW command) to device and switch execution from FW2 or primary image. And the device will be again not available till we complete boot sequence.

This message is used to notify user that dock will not be available for this duration.